opflexagent/test/test_agent_types.py

```diff
@@ -38,10 +38,11 @@ def test_opflex_agent_mode(self):
             'opflexagent.as_metadata_manager.AsMetadataManager')
         metadata_patch = metadata_mgr.start()
         cfg.CONF.set_override('agent_mode', 'opflex', 'OPFLEX')
-        with mock.patch('sys.argv'):
-            gbp_agent.main()
-            self.assertEqual(1, opflex_patch.call_count)
-            self.assertEqual(1, metadata_patch.call_count)
+        with mock.patch('os.path.basename'):
```

Comment:
This is a general comment on replacing contextlib.nested patterns, which is used throughout this patch (although this line didn't have it originally, this fix for python3 would benefit from this pattern if it were available). In python3, it's possible to just use a comma-delimited list of statements that need a context manager, instead of cascading them like this (see https://books.google.com/books?id=flRFgvAU7N8C&pg=PA370&lpg=PA370&dq=python3+contextlib+nested+replacement&source=bl&ots=pFTexqEi0Q&sig=ACfU3U1eLdrdddgK5PRWQ-dO0ss7ZBe5ZA&hl=en&sa=X&ved=2ahUKEwjrwuq-qYLoAhXJGM0KHfbeBvEQ6AEwCXoECAoQAQ#v=onepage&q=python3%20contextlib%20nested%20replacement&f=false and https://stackoverflow.com/questions/16083791/alternative-to-contextlib-nested-with-variable-number-of-context-managers). I think these alternatives would read better than the cascaded approach.

Reply:
we have removed nested Mock and we used ExitStack in place of that.